### DISCUSSION TITLE
Does anyone run Frigate OpenVino on an Intel Arc graphics card?

### DISCUSSION THREAD

**Discussion by WayneManion** [blakeblackshear/frigate#7354] [Ask A Question] [answered]
The title pretty much says it all. I have a Xeon system with an AMD graphics card running Linux. I won't buy an Nvidia card and I have been waiting forever and a day to be able to buy a Coral USB for MSRP. 

Using an Intel A380 seems like an interesting alternative. Is anyone running Frigate OpenVino on one of these cards? In a VM with PCIe passthrough?

Thanks

**Comment by noloboy** [comment]
Were you able to get an answer to this?  I'm on the same boat and wanted to know how intel arc gpus work with frigate both on transcoding and detection (open VINO)

**Comment by WayneManion** [comment]
I never did get an authoritative answer. I bought an Arc A380 for my Xeon box when it was on sale. I honestly wasn't  sure if the card would work in my machine at all because my old motherboard doesn't support resizable BAR. The card works fine in my computer, but I haven't had any time/motivation to see if Frigate will work with it.

**Reply by NickM-27** [reply]
I don't see any reason it wouldn't. If the driver sees it then it's no different to frigate.

**Reply by Dvalin21** [reply]
Were you able to get this to work?

**Reply by WayneManion** [reply]
I was able to source a Google Coral for detection. I tried to use the Arc card for video acceleration, but encountered consistent crashes when enabling acceleration. 

I ended up moving the Google Coral to another machine with Intel 8th Gen CPU with integrated graphics. That setup has worked flawlessly.

**Reply by Dvalin21** [reply]
Thanks for the reply

**Reply by Dvalin21** [reply]
> I was able to source a Google Coral for detection. I tried to use the Arc card for video acceleration, but encountered consistent crashes when enabling acceleration.
> 
> I ended up moving the Google Coral to another machine with Intel 8th Gen CPU with integrated graphics. That setup has worked flawlessly.

I found something just if you were curious https://community.intel.com/t5/Intel-Distribution-of-OpenVINO/Can-we-run-openvino-toolkit-on-AMD-CPU-with-Intel-Arc-770-GPU/m-p/1553103  I know you didn't have a AMD CPU, but apparently only certain Xeon CPUs are compatible? Not sure which one you had.

**Reply by WayneManion** [reply]
I was using a Xeon E5-2697 v4. This is a Broadwell-based (~2016) 18-core part with no integrated graphics. Some other series of Xeons are more similar to desktop chips and have integrated GPUs. 

I was running the Xeon on an Asus X99 motherboard with no support for resizable bar (rebar). Intel's Arc graphics cards rely pretty heavily on rebar. 

On Windows the performance in games drops like a rock on platforms without rebar support. The Linux driver may not be able to compensate for lack of rebar at all. The Linux Arc graphics driver just might not be very stable. I don't know. I know that all of the temperature monitoring features of Arc are totally missing on Linux. I realize that old Xeon on X99 with Intel Arc graphics running Linux is not a very common combination.

**Reply by Dvalin21** [reply]
Thanks for the info!!

**Comment by mrplow** [comment]
I'm planning on getting an Arc card for transcoding in my next server build later this year. I have a coral for detection so hopefully I won't run into any issues.
I don't see why it wouldn't work. Arc cards should work the same with openvino as Intel CPUs.

**Comment by Timdawson264** [comment]
The main issue I have encountered thus far is that Debian 11 does not have OpenCL support for intel DG2 (arc).
with openvino on Debian 12 there should be no issues running frigate detection, if i get time i will build an image to test it out.

```
root@frigate-756c974bdf-bn8jk:/opt/frigate# apt info intel-opencl-icd
Package: intel-opencl-icd
Version: 22.43.24595.41-1
Status: install ok installed
Priority: optional
Section: libs
Source: intel-compute-runtime
Maintainer: Debian OpenCL Maintainers <pkg-opencl-devel@lists.alioth.debian.org>
Installed-Size: 22.7 MB
Provides: opencl-icd
Depends: libc6 (>= 2.34), libgcc-s1 (>= 3.4), libigdgmm12 (>= 22.3.2+ds1), libstdc++6 (>= 12), libigc1 (>= 1.0.12504.6), libigc1 (<< 1.0.12504.6+~), libigc1+llvm14, libigdfcl1 (>= 1.0.12504.6), libigdfcl1 (<< 1.0.12504.6+~), libigdfcl1+llvm14, ocl-icd-libopencl1
Breaks: intel-opencl
Replaces: intel-opencl
Homepage: https://github.com/intel/compute-runtime
Download-Size: unknown
APT-Manual-Installed: yes
APT-Sources: /var/lib/dpkg/status
Description: Intel graphics compute runtime for OpenCL
 The Intel(R) Graphics Compute Runtime for OpenCL(TM) is an open source
 project to converge Intel's development efforts on OpenCL(TM) compute
 stacks supporting the GEN graphics hardware architecture.
 .
 Supported platforms - OpenCL 3.0, Production:
  - DG1
  - Intel Core Processors with Gen9 GPU (Skylake)
  - Intel Core Processors with Gen9.5 GPU (Kaby Lake, Coffee Lake)
  - Intel Core Processors with Gen11 GPU (Ice Lake)
  - Intel Core Processors with Xe LP (Gen12) GPU (Tiger Lake, Rocket Lake)
  - Intel Core Processors with Xe (Gen12.2) GPU (Alder Lake)
  - Intel Atom Processors with Gen11 GPU (Elkhart Lake)
 Supported platforms - OpenCL 3.0, Maintenance:
  - Intel Core Processors with Gen8 GPU (Broadwell)
  - Intel Atom Processors with Gen9 GPU (Apollo Lake, Gemini Lake)

#
apt update && apt install clinfo
clinfo
Number of platforms                               0

```

**Comment by WayneManion** [comment]
I think Arc support on Linux is pretty threadbare, at least on platforms without rebar support. I have another X99 machine with a no-iGPU Broadwell Xeon. I run Arch desktop on this machine with Gnome. I had an Arc 380 in it. I would get crashes in Bambu Studio and Orca Slicer pretty regularly. I would also get constant complaints about OpenGL (or web GL, I can't remember) when I ran OnShape in the browser.

I swapped the card out for an AMD Radeon RX 5500 and all of those problems with Bambu Studio, Orca Slicer, and OnShape disappeared. 

I wouldn't _expect_ anything other than basic desktop support to work properly on an Arc card under Linux right now. Maybe things are much better on systems with rebar support. 

I got constant system crashes just trying to get video acceleration to work. I never even tried OpenVino. I already had a better-supported Google Coral device for object detection.

**Comment by mic-s** [comment]
I was able to run OpenVino and hardware acceleration on Intel Arc A380. I have done it in proxmox LXC container with Ubuntu 24, on default docker with installed packages in docker container:
```
docker exec -ti frigate bash
cd $(mktemp -d)

wget https://github.com/intel/intel-graphics-compiler/releases/download/igc-1.0.16510.2/intel-igc-core_1.0.16510.2_amd64.deb
wget https://github.com/intel/intel-graphics-compiler/releases/download/igc-1.0.16510.2/intel-igc-opencl_1.0.16510.2_amd64.deb
wget https://github.com/intel/compute-runtime/releases/download/24.13.29138.7/intel-level-zero-gpu-dbgsym_1.3.29138.7_amd64.ddeb
wget https://github.com/intel/compute-runtime/releases/download/24.13.29138.7/intel-level-zero-gpu_1.3.29138.7_amd64.deb
wget https://github.com/intel/compute-runtime/releases/download/24.13.29138.7/intel-opencl-icd-dbgsym_24.13.29138.7_amd64.ddeb
wget https://github.com/intel/compute-runtime/releases/download/24.13.29138.7/intel-opencl-icd_24.13.29138.7_amd64.deb
wget https://github.com/intel/compute-runtime/releases/download/24.13.29138.7/libigdgmm12_22.3.18_amd64.deb

dpkg -i *.deb
```


Interference speed was around 6ms on single camera with 3072x2048 resolution

On Debian 12 LXC container I was able only to run hardware acceleration.

**Reply by Timdawson264** [reply]
did you build an ubuntu24 frigate image?

**Reply by mic-s** [reply]
No, I have installed ubuntu packages on default frigate docker container, they needed to be repacked, but it worked.
So it was Proxmox (kernel 6.8.12-1) > Ubuntu 24 in LXC > Frigate Debian docker.
At beginning there was known problem with missing packages, so I followed steps from:
https://github.com/blakeblackshear/frigate/discussions/12012#discussioncomment-9794463
and then I was able to run openvino. I am not sure if different system in LXC is making difference because I stopped checking when hardware acceleration was working on my "production" LXC with Debian 12

**Comment by NickM-27** [comment, marked as answer]
A user posted with this working on reddit https://www.reddit.com/r/frigate_nvr/comments/1f7yi9o/frigate_with_openvino_on_intel_arc_a380/?rdt=44320

**Reply by mesencephalon** [reply]
I also made it work with Intel Arc Pro A40 GPU utilizing OpenVino. It takes a lot of work though requiring some package updates in Frigate docker container

**Reply by NickM-27** [reply]
that is most likely resolved as of the commits in dev today

**Reply by mesencephalon** [reply]
Thanks for the update, much appreciated. I use UNRAID and Frigate is a docker there. It means when I have a Frigate update next time in UNRAID through the community apps, I guess I'll not have to go through all the hassle I went through with package updates, is that correct?

**Reply by NickM-27** [reply]
presumably, but there is no reason you need to wait for that. I use unraid too, you an always just change the image in the config ot be a different image
